asreviewcontrib/datatools/compose.py

```diff
@@ -0,0 +1,255 @@
+    allowed_orders = ["riu", "rui", "uri", "uir", "iru", "iur"]
+    if order not in allowed_orders:
+        raise ValueError(
+            f"An unsupported order was given with --priority, choose one of the following: {allowed_orders}"
+        )
+
+    return
+
+
+def _check_resolve_arg(resolve):
+    allowed_resolve = ["continue", "abort", "keep"]
+    if resolve not in allowed_resolve:
+        raise ValueError(
+            f"An unsupported method for conflict resolving was given with --conflict_resolve, choose one "
+            f"of the following: {allowed_resolve}"
+        )
+
+    return
+
+
+def _check_suffix(input_files, output_file):
+    # Also raises ValueError on URLs that do not end with a file extension
+    suffixes = [Path(item).suffix for item in input_files if item is not None]
+    suffixes.append(Path(output_file).suffix)
+
+    set_ris = {'.txt', '.ris'}
+    set_tabular = {'.csv', '.tab', '.tsv', '.xlsx'}
+    set_suffixes = set(suffixes)
+
+    if len(set(suffixes)) > 1:
+        if not (set_suffixes.issubset(set_ris) or set_suffixes.issubset(set_tabular)):
+            raise ValueError(
+                "Several file types were given, all input files as well as the output file should be of the same type."
+            )
+
+
+def _check_label_errors(as_lab, path_lab):
+    if as_lab is not None:
+        if as_lab.labels is None:
+            warnings.warn(
+                f"No labels were found in '{path_lab}', continuing with its records marked as "
+                f"unlabeled. If this is not correct, check if your data format complies with: "
+                f"https://asreview.readthedocs.io/en/latest/data_format.html"
+            )
+
+    return
+
+
+def _append_df(list_df, as_obj, label):
+    # retrieve part of dataframe with label -1, 0 or 1
+    df_slice = as_obj.df[as_obj.labels == label].reset_index(drop=True)
+
+    if not df_slice.empty:
+        list_df.append(df_slice)
+
+    return
+
+
+def _concat_label(list_df, label, pid="doi"):
+    # if there are any dataframes with the given label, concatenate and drop duplicates on pid and title/abstract
+    if list_df:
+        df_all = pd.concat(list_df).reset_index(drop=True)
+        df_all["included"] = label
+        n_total = len(df_all)
+
+        df_all = ASReviewData(df=df_all).drop_duplicates(pid=pid).reset_index(drop=True)
+
+        n_total_dedup = n_total - len(df_all)
+        print(
+            f"Detected {n_total} records with label '{label}', from which {n_total_dedup} duplicate records with the "
+            f"same label were removed."
+        )
+    else:
+        df_all = pd.DataFrame()
+
+    return df_all
+
+
+def create_composition(input_files, pid="doi", order="riu", resolve="continue"):
+    # load all input files and URLs into ASReviewData objects, fill with None if input was not specified
+    as_rel, as_irr, as_lab, as_unl = [
+        load_data(item) if item is not None else None for item in input_files
+    ]
```

Comment:
Explicit is better than implicit (`import zen`). I would propose replacing the input files with four file paths.